### PR TITLE
H2 integration

### PR DESCRIPTION
**Pull request by warfawl**
added dependencies for `spring data` and `H2`
added H2 profile
created `User` entity class and repository
created DDL scripts

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -1,2 +1,3 @@
-server:
```
Comment (zozich) [inline src/main/resources/application.yml]: Why did you remove `server.port`?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
-  port: 9000
+spring:
+  profiles:
+    active: h2
```
Comment (zozich) [inline src/main/resources/application.yml]: Call active profile something like `dev`

File: src/test/java/ua/danit/final_project/FinalProjectApplicationTests.java [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.junit4.SpringRunner;
+import ua.danit.final_project.repositories.UserRepo;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
 public class FinalProjectApplicationTests {
 
+  @Autowired
+  UserRepo userRepo;
```
Comment (zozich) [inline src/test/java/ua/danit/final_project/FinalProjectApplicationTests.java]: Do not autowire fields. Autoworing should better be done on a constructor level

**Comment by warfawl** [comment]
[Use H2 database for local development](https://trello.com/c/554Skxhl/14-use-h2-database-for-local-development)
